src/webhook/mutation/pod_mutator/init_container.go

```diff
@@ -44,34 +43,59 @@ func securityContextForInitContainer(pod *corev1.Pod) *corev1.SecurityContext {
 				"ALL",
 			},
 		},
-		SeccompProfile: &corev1.SeccompProfile{
-			Type: corev1.SeccompProfileTypeRuntimeDefault,
-		},
 	}
 
+	copySecurityContextUserAndGroup(securityContext, *pod)
+
+	return securityContext
+}
+
+func copySecurityContextUserAndGroup(securityContext *corev1.SecurityContext, pod corev1.Pod) {
 	containerSecurityContext := pod.Spec.Containers[0].SecurityContext
 	podSecurityContext := pod.Spec.SecurityContext
 
-	if hasContainerUserAndGroupSet(containerSecurityContext) {
-		securityContext.RunAsGroup = containerSecurityContext.RunAsGroup
-		securityContext.RunAsUser = containerSecurityContext.RunAsUser
-	} else if hasPodUserAndGroupSet(podSecurityContext) {
-		securityContext.RunAsGroup = podSecurityContext.RunAsGroup
+	securityContext.RunAsUser = address.Of(defaultUser)
+	securityContext.RunAsGroup = address.Of(defaultGroup)
+
+	if hasPodUserSet(podSecurityContext) {
 		securityContext.RunAsUser = podSecurityContext.RunAsUser
-	} else {
-		securityContext.RunAsGroup = address.Of(int64(1001))
-		securityContext.RunAsUser = address.Of(int64(1001))
+	}
+	if hasPodGroupSet(podSecurityContext) {
+		securityContext.RunAsGroup = podSecurityContext.RunAsGroup
 	}
 
-	return securityContext
+	if hasContainerUserSet(containerSecurityContext) {
+		securityContext.RunAsUser = containerSecurityContext.RunAsUser
+	}
+	if hasContainerGroupSet(containerSecurityContext) {
+		securityContext.RunAsGroup = containerSecurityContext.RunAsGroup
+	}
+
+	if isNonRoot(securityContext) {
+		securityContext.RunAsNonRoot = address.Of(true)
+	}
+}
+
+func hasPodUserSet(ctx *corev1.PodSecurityContext) bool {
+	return ctx != nil && ctx.RunAsUser != nil
+}
+
+func hasPodGroupSet(ctx *corev1.PodSecurityContext) bool {
+	return ctx != nil && ctx.RunAsGroup != nil
+}
+
+func hasContainerUserSet(ctx *corev1.SecurityContext) bool {
+	return ctx != nil && ctx.RunAsUser != nil
 }
 
-func hasContainerUserAndGroupSet(ctx *corev1.SecurityContext) bool {
-	return ctx != nil && ctx.RunAsUser != nil && ctx.RunAsGroup != nil
+func hasContainerGroupSet(ctx *corev1.SecurityContext) bool {
+	return ctx != nil && ctx.RunAsGroup != nil
 }
 
-func hasPodUserAndGroupSet(ctx *corev1.PodSecurityContext) bool {
-	return ctx != nil && ctx.RunAsUser != nil && ctx.RunAsGroup != nil
+func isNonRoot(ctx *corev1.SecurityContext) bool {
+	return ctx != nil &&
+		(ctx.RunAsUser != nil && *ctx.RunAsUser != rootUser) &&
+		(ctx.RunAsGroup != nil && *ctx.RunAsGroup != rootUser)
```

Comment:
`rootUser` constant is used for both "user" and "group", so it should be named to something that better reflects that